opal/mca/btl/tcp/btl_tcp_endpoint.c

```diff
@@ -1101,7 +1101,9 @@ static void mca_btl_tcp_endpoint_send_handler(int sd, short flags, void* user)
             /* if required - update request status and release fragment */
             OPAL_THREAD_UNLOCK(&btl_endpoint->endpoint_send_lock);
             assert( frag->base.des_flags & MCA_BTL_DES_SEND_ALWAYS_CALLBACK );
-            frag->base.des_cbfunc(&frag->btl->super, frag->endpoint, &frag->base, frag->rc);
+            if (NULL != frag->base.des_cbfunc) {
```

Comment:
should this be in a later PR or its own PR?

Reply:
Moved it into its own commit. Needed for this PR because the tcp btl force sets this flag ignoring the intent of the caller. It is reasonable to expect the caller doesn't care when the frag was actually sent so a NULL callback is reasonable.